### PR TITLE
Added support for Files Change Time

### PR DESCRIPTION
**Pull request by seanmcc-msft**

### CODE REVIEW

File: sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs [hunk 1]
```diff
@@ -714,6 +714,7 @@ private void SetNameFieldsIfNull()
                             fileAttributes: smbProps.FileAttributes?.ToAttributesString() ?? Constants.File.FileAttributesNone,
                             fileCreationTime: smbProps.FileCreatedOn.ToFileDateTimeString() ?? Constants.File.FileTimeNow,
                             fileLastWriteTime: smbProps.FileLastWrittenOn.ToFileDateTimeString() ?? Constants.File.FileTimeNow,
+                            fileChangeTime: smbProps.FileChangedOn.ToFileDateTimeString(),
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs:717]: Would this get set to `default` if there's no value? Just making sure it doesn't get set to something weird like `DateTimeOffset.MinValue` or something.

File: sdk/storage/Azure.Storage.Files.Shares/src/autorest.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ Run `dotnet build /t:GenerateCode` to generate code.
 
 ``` yaml
 input-file:
-    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/d710fb7f40338ca6ef00e0a9680be0a3d4cf5059/specification/storage/data-plane/Microsoft.FileStorage/preview/2021-04-10/file.json
+    - C:\Users\seanmcc\git\azure-rest-api-specs\specification\storage\data-plane\Microsoft.FileStorage\preview\2021-06-08\file.json
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Files.Shares/src/autorest.md:7]: Remember to revert back

File: sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs (hunk 1 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs:717]: This parameter is optional, with a default value set to the time of the request time.  [Spec](https://microsoft.sharepoint.com/:w:/t/AzureStorage/EWulSSFYq0hEuvhA2RMI0IgBdYZiKrpALCxYXZJ5P7nXzQ) for more context.

**Review by amnguye — APPROVED** [review]

File: sdk/storage/Azure.Storage.Files.Shares/tests/ShareClientTestFixtureAttribute.cs [hunk 3]
```diff
@@ -25,7 +25,7 @@ public ShareClientTestFixtureAttribute()
                 StorageVersionExtensions.MaxVersion)
         {
             RecordingServiceVersion = StorageVersionExtensions.MaxVersion;
-            LiveServiceVersions = new object[] { StorageVersionExtensions.LatestVersion, };
+            LiveServiceVersions = new object[] { StorageVersionExtensions.MaxVersion, };
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.Shares/tests/ShareClientTestFixtureAttribute.cs:28]: TODO revert this.

**Review by jaschrep-msft — APPROVED** [review]

**Pull request merged**